src/compiler/core.ts

```diff
@@ -373,11 +373,11 @@ namespace ts {
         return false;
     }
 
-    export function filterMutate<T>(array: T[], f: (x: T) => boolean): void {
+    export function filterMutate<T>(array: T[], f: (x: T, i: number) => boolean): void {

```

Comment:
If you align more closely with `Array.prototype.filter` and use `(x: T, i: number, array: T[]) => boolean`, then you can remove the [closure](https://github.com/Microsoft/TypeScript/pull/17248/files#diff-21767a350f9d093aef26a0497b4735c7R206) in `toDeduplicatedSortedArray` by extracting it into a reusable function:

```ts
function isNonDuplicate<T>(value: T, index: number, array: T[]) {
  return i === 0 || value !== array[index - 1]; 
} 
```